packages/components/dropdowns/dropdown-menu/src/dropdown-menu.spec.tsx

```diff
@@ -0,0 +1,65 @@
+import SecondaryButton from '@commercetools-uikit/secondary-button';
+
+import { screen, render } from '../../../../../test/test-utils';
+import DropdownMenu from './dropdown-menu';
+
+describe('DropdownMenu', () => {
+  it('should render list menu', async () => {
+    const firstOptionOnClick = jest.fn();
+    const secondOptionOnClick = jest.fn();
+
+    await render(
+      <DropdownMenu
+        menuType="list"
+        triggerElement={<SecondaryButton label="Trigger" />}
+      >
+        <DropdownMenu.ListMenuItem onClick={firstOptionOnClick}>
+          Option 1
+        </DropdownMenu.ListMenuItem>
+        <DropdownMenu.ListMenuItem onClick={secondOptionOnClick} isDisabled>
+          Option 2
+        </DropdownMenu.ListMenuItem>
+      </DropdownMenu>
+    );
+
+    // Dropdown shold be closed by default
+    expect(screen.queryByText('Option 1')).not.toBeVisible();
+
+    // Open the dropdown
+    screen.getByLabelText('Trigger').click();
+    expect(await screen.findByText('Option 1')).toBeVisible();
+    expect(screen.getByText('Option 2')).toBeVisible();
+
+    // Clicking in the disabled options should do nothing
+    screen.getByText('Option 2').click();
+    expect(secondOptionOnClick).toHaveBeenCalledTimes(0);
+    expect(await screen.findByText('Option 1')).toBeVisible();
+
+    // Clicking in the enabled option should close the dropdown
+    screen.getByText('Option 1').click();
+    expect(firstOptionOnClick).toHaveBeenCalledTimes(1);
+    expect(await screen.findByText('Option 1')).not.toBeVisible();
+  });
+
+  it('should render the default menu', async () => {
+    await render(
+      <>
+        <h2>Header</h2>
+        <DropdownMenu triggerElement={<SecondaryButton label="Trigger" />}>
+          <div>Content</div>
+        </DropdownMenu>
+      </>
+    );
+
+    // Dropdown shold be closed by default
+    expect(screen.queryByText('Content')).not.toBeVisible();
+
+    // Open the dropdown
+    screen.getByLabelText('Trigger').click();
+    expect(await screen.findByText('Content')).toBeVisible();
+
+    // Clicking outside the dropdown should close it
+    screen.getByText('Header').click();
+    expect(await screen.findByText('Content')).not.toBeVisible();
```

Comment:
```suggestion
  await act(async () => {
      // Clicking outside the dropdown should close it
    screen.getByText('Header').click();
    expect(await screen.findByText('Content')).not.toBeVisible();
    });
```

I was able to clear the error by using the suggested act to wait for the state to be completed before asserting, from what I found.

Reply:
`import { act } from 'react-dom/test-utils';`

Reply:
fair point. 
I saw some js tests using it although act imported from a different package.

Maybe another alternative could to waitFor.

```suggestion
    await waitFor(() => {
      screen.getByText('Header').click();
      // Check if Content is not visible
      const contentElement = screen.queryByText('Content');
      expect(contentElement).not.toBeVisible();
    });
```

Reply:
I also tried that one but didn't work in my machine.
Did it work for you?

Reply:
Yes, it works.

Reply:
<img width="773" alt="image" src="https://github.com/commercetools/ui-kit/assets/19975842/e2f15452-a141-4942-ba8e-248f051e61a8">

Reply:
But bear in mind the tests won't fail locally. They will fail on CI because over there we raise an error if we use `console.error`.

Maybe you can create another PR our of this one with that change and check if it works over there, because it failed when I tried it.

Reply:
Okay, let me try that now.

Reply:
Looks like its just the vrt that's failing now. But I cannot see that error on this PR: https://github.com/commercetools/ui-kit/pull/2771